File: server/php/src/Presentation/Models/Payment/Response.php
<?php

namespace OnlinePayments\ExampleApp\Presentation\Models\Payment;

use OnlinePayments\ExampleApp\Application\Domain\Enums\AdditionalPaymentActions\Status;
use OnlinePayments\ExampleApp\Application\Domain\Payments\StatusOutput;
use OnlinePayments\ExampleApp\Application\Interfaces\Responses\ResponseInterface;

class Response implements ResponseInterface
{
    public ?string $id;
    public ?Status $status;
    public ?StatusOutput $statusOutput;

    public function __construct(
        ?string $id = null,
        ?Status $status = null,
        ?StatusOutput $statusOutput = null
    ) {
        $this->id = $id;
        $this->status = $status;
        $this->statusOutput = $statusOutput ?? new StatusOutput();
    }

    public function toArray(): array
    {
        return [
            'id' => $this->id,
            'status' => $this->status?->value,
            'statusOutput' => $this->statusOutput ? [
                'statusCode' => $this->statusOutput->statusCode,
                'statusCategory' => $this->statusOutput->statusCategory?->value
            ] : null
        ];
    }
}

File: server/ruby/app/presentation/models/additional_payment_actions/response.rb
module Presentation
  module Models
    module AdditionalPaymentActions
      class Response
        attr_accessor :id, :status, :status_output

        def initialize(id: nil, status: nil, status_output: nil)
          @id = id
          @status = status
          @status_output = status_output
        end

        def to_h
          {
            id: id,
            status: status&.to_s,
            statusOutput: status_output ? {
              statusCode: status_output.status_code,
              statusCategory: status_output.status_category
            } : nil
          }
        end
      end
    end
  end
end
